testsupport/tiers/nstemplateset.go

```diff
@@ -142,6 +143,42 @@ func UntilNSTemplateSetHasTemplateRefs(expectedRevisions TemplateRefs) wait.NSTe
 	}
 }
 
+// UntilNSTemplateSetHasStatusTemplateRefs checks if the NSTemplateTier has the expected template refs in the Status
+func UntilNSTemplateSetHasStatusTemplateRefs(expectedRevisions TemplateRefs) wait.NSTemplateSetWaitCriterion {
+	return wait.NSTemplateSetWaitCriterion{
+		Match: func(actual *toolchainv1alpha1.NSTemplateSet) bool {
+			if expectedRevisions.ClusterResources == nil ||
+				actual.Status.ClusterResources == nil ||
+				*expectedRevisions.ClusterResources != actual.Status.ClusterResources.TemplateRef {
+				return false
+			}
+			actualNamespaceTmplRefs := make([]string, len(actual.Status.Namespaces))
+			for i, r := range actual.Status.Namespaces {
+				actualNamespaceTmplRefs[i] = r.TemplateRef
+			}
+			sort.Strings(actualNamespaceTmplRefs)
+			sort.Strings(expectedRevisions.Namespaces)
+			if !reflect.DeepEqual(actualNamespaceTmplRefs, expectedRevisions.Namespaces) {
+				return false
+			}
+
+			// check expected feature toggles, if any
+			featureAnnotation, featureAnnotationFound := actual.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey]
+			if featureAnnotationFound {
+				if !reflect.DeepEqual(utils.SplitCommaSeparatedList(featureAnnotation), actual.Status.FeatureToggles) {
+					return false
+				}
+			}
+
+			// checks that the actual spaceroles are at least subSet of the expected spaceroles ones
+			return IsSpaceRoleSubset(expectedRevisions.SpaceRoles, actual.Status.SpaceRoles)
```

Comment:
this is actually the place where we should really verify that the applied SpaceRoles (that are set in the status) exactly match the spec, and not the subset of the expected revisions.

Reply:
done in https://github.com/codeready-toolchain/toolchain-e2e/commit/db0fa2bcc320561a41b48289f421609c9d3f96ea